aten/src/ATen/native/quantized/cpu/qconcat.cpp

```diff
@@ -0,0 +1,51 @@
+#include <ATen/ATen.h>
+#include <ATen/core/Type.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cpu/Loops.h>
+
+#include <algorithm>
+#include <vector>
+
+namespace at { namespace native {
+namespace {
+template <bool ReLUFused = false>
+class QCat final : public torch::OperatorKernel {
+ public:
+  Tensor operator()(const std::vector<Tensor>& qxs, int64_t axis,
+                    double scale, int64_t zero_point) {
+    const auto x_dtype = qxs[0].scalar_type();
```

Comment:
Lets make scale and zero-point optional. If they are not specified, the operator should calculate a default scale and zero-point.
1. First calculate the ranges(min_value, max_value) for every tensor: This is [ (range_min -zero_point)*scale, (range_max -zero_point)*scale], range_min =0 and range_max=255 for quint8 and range_min = -128, range_max = 127 for qint8.
2. Pick a = min(min_values) and b = max(max_values)
3. Calculate scale as (b-a)/255
4. Calculate zero point as: range_min - floor(a/scale)

Reply:
will this be useful?

Reply:
Should I add the optional stuff on the cpp level or in the functional/module level?